### PR TITLE
All exercises: Changed the structure of the exercises to better align with course curriculum

### LINKED ISSUE #593: All exercises: Restucture folders so they reflect the structure of the curriculum
Complete the following REQUIRED checkboxes:
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this issue follows the `location for request: brief description of request` format, e.g. `Foundations exercises: Add exercise for XYZ`

The following checkbox is OPTIONAL:

-   [ ] I would like to be assigned this issue to work on it

<hr>

**1. Description of the Feature Request:**

There are multiple users (from the Discord) that accidentally worked through more advanced exercises. I believe that structuring the `css-exercises` repo similar to the curriculum's structure would serve as a bit of a "fail-safe".

It will also make it easier for everyone else to navigate to the correct exercises.

i.e.
Instead of the folders being laid out like they currently are:
```bash
animation  flex  foundations  grid  margin-and-padding
```
we can structure them just like they appear in the curriculum:
```bash
foundations  intermediate-html-and-css  advanced-html-and-css
```
and the nesting can go deeper if deemed necessary.

However, the drawback is that any potential, future updates to the curriculum's structure will require an update in the `css-exercises` repo as well. This will, in a way, "couple" this repo's layout to the curriculum's layout.

**2. Acceptance Criteria:**

 N/A

**3. Additional Information:**

Relevant PR that helped address this concern: https://github.com/TheOdinProject/curriculum/pull/28754

I prefer that a first-time/new contributor takes this one. If no one volunteers after some time, I will likely ask to be assigned to this if the suggestion is accepted.

### PR DESCRIPTION
**Pull request by oceanthunder**
## Because

Gives a better structure to the exercises by dividing the exercises as they occur in TOP.
```
├── advanced-html-css
│   └── animation
│       ├── 01-button-hover
│       ├── 02-pop-up
│       └── 03-dropdown-menu
├── foundations
│   ├── block-and-inline
│   │   ├── 01-margin-and-padding-1
│   │   └── 02-margin-and-padding-2
│   ├── cascade
│   │   └── 01-cascade-fix
│   ├── flex
│   │   ├── 01-flex-center
│   │   ├── 02-flex-header
│   │   ├── 03-flex-header-2
│   │   ├── 04-flex-information
│   │   ├── 05-flex-modal
│   │   ├── 06-flex-layout
│   │   └── 07-flex-layout-2
│   └── intro-to-css
│       ├── 01-css-methods
│       ├── 02-class-id-selectors
│       ├── 03-grouping-selectors
│       ├── 04-chaining-selectors
│       └── 05-descendant-combinator
└── intermediate-html-css
    ├── advanced-grid
    │   ├── 01-responsive-holy-grail
    │   └── 02-holy-grail-mockup
    └── positioning-grid
        └── 01-basic-holy-grail
```

## This PR

* It divides the exercises in 3 directories

## Issue

Closes #593 

## Additional Information

No

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `01-flex-center: Update self check`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If applicable, I have ensured that the TOP solution files match the Desired Outcome image

### CODE REVIEW

**Review by MaoShizhong — APPROVED** [review]
Thanks for doing this.

Merge blocked until respective curriculum repo PR has been approved, as detailed in #593.

**Comment by oceanthunder** [comment]
> Thanks for doing this.
No, thank you!

**Pull request merged**
